docs/models/clip.md

```diff
@@ -0,0 +1,128 @@
+conda env create -f conda_env_torch_zoo.yml
+```
+
+---
+
+## Minimal Working Example
+
+[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/drive/13BamfEryOz0z_M-zWw5MHdf4lVvBtwj0?usp=sharing)
+
+Activate the environment
+```bash
+conda activate torch_zoo
+```
+
+and extract features at 1 fps from `./sample/v_GGSY1Qvo990.mp4` video and output results.
+```bash
+python main.py \
+    feature_type=clip \
+    model_name=ViT-B/32 \
+    extraction_fps=1 \
+    video_paths="[./sample/v_GGSY1Qvo990.mp4]" \
+    on_extraction=print
+```
+
+---
+
+## Examples
+Start by activating the environment
+```bash
+conda activate torch_zoo
+```
+
+It is pretty much the same procedure as with other features. 
+Here we take `ViT/B-32` as an example, but we also support `ViT-B/16`, `RN50x16`, `RN50x4`, `RN101`, `RN50` and others in [openAI CLIP implementation](https://github.com/openai/CLIP).
+In addition, if you want to use your weights, you need to copy your weights to `models/clip/checkpoints`, rename it `CLIP-custom.pth` and specify `model_name=custom`.
+```bash
+python main.py \
+    feature_type=clip \
+    model_name=ViT-B/32 \
+    device_ids="[0, 2]" \
+    video_paths="[./sample/v_ZNVhz7ctTq0.mp4, ./sample/v_GGSY1Qvo990.mp4]"
+```
+If you would like to save the features, use `--on_extraction save_numpy` (or `save_pickle`) – by default, the features are saved in `./output/` or where `--output_path` specifies. In the case of frame-wise features, besides features, it also saves timestamps in ms and the original fps of the video into the same folder with features.
+```bash
+python main.py \
+    feature_type=clip \
+    model_name=ViT-B/32 \
+    device_ids="[0, 2]" \
+    on_extraction=save_numpy \
+    file_with_video_paths=./sample/sample_video_paths.txt
+```
+We may increase the extraction speed with batching. Therefore, frame-wise features have `--batch_size` argument, which defaults to `1`.
+```bash
+python main.py \
+    feature_type=clip \
+    model_name=ViT-B/32 \
+    device_ids="[0, 2]" \
+    batch_size=128 \
+    video_paths="[./sample/v_ZNVhz7ctTq0.mp4, ./sample/v_GGSY1Qvo990.mp4]"
+```
+If you would like to extract features at a certain fps, add `--extraction_fps` argument
+```bash
+python main.py \
+    feature_type=clip \
+    model_name=ViT-B/32 \
+    device_ids="[0, 2]" \
+    extraction_fps=5 \
+    video_paths="[./sample/v_ZNVhz7ctTq0.mp4, ./sample/v_GGSY1Qvo990.mp4]"
+```
+
+If you would like to verify the extracted features, you can
+set `show_pred=true` and provide several sentences with `pred_texts` argument.
+The value of `pred_texts` should be a list of strings. 
+The probability that each frame corresponds to all the sentences you provide will be output.
+```bash
+python main.py \
+    feature_type=clip \
+    model_name=ViT-B/32 \
+    device_ids="[0, 2]" \
+    extraction_fps=1 \
+    pred_texts="['a dog smiles', 'a man is lifting']" \
```

Comment:
TODO for myself: it feels like we could still benefit from showing zero-shot on kinetics classes if `pred_texts` is not defined.